### ISSUE TITLE
cleanup 2.0: remove Engagement.eng_type field + table

### ISSUE THREAD

**Issue by valentijnscholten** [DefectDojo/django-DefectDojo#3728]
looks as having been replaced by engagement_type field

**Comment by kiblik** [comment]
+1

It is not possible to use API (`GET /engagements/`) to list only `CI/CD` or only `Interactive` engagements now.
API support only `eng_type(string)` and it returns
```
{
  "eng_type": [
    "Select a valid choice. That choice is not one of the available choices."
  ]
}
```
when I use one of valid values

**Comment by valentijnscholten** [comment]
You need to lookup the id column in the Engagement_Type table.

**Comment by kiblik** [comment]
But it is not accessible from APIv2, isn't it?

**Comment by damiencarol** [comment]
@kiblik it is accessible in API v2. (we use it at my company to manage CI/CD engagements dynamically)
![image](https://user-images.githubusercontent.com/1694940/116069137-59ab5e00-a68b-11eb-86c4-e3a4553f5dc2.png)

You just push the engagement with attribute `engagement_type` to `Interactive` or `CI/CD`

**Comment by kiblik** [comment]
But this `engagement_type` is missing in simple `GET /api/v2/engagements/`.

![image](https://user-images.githubusercontent.com/5609770/116259515-c6972480-a76d-11eb-860d-266812443dd2.png)

And using `CI/CD` in `eng_type` is not possible:
```json
curl -X GET "https://.../api/v2/engagements/?eng_type=CI%2FCD" ...
{
  "eng_type": [
    "Select a valid choice. That choice is not one of the available choices."
  ]
}
```

**Comment by valentijnscholten** [comment]
You need to use `engagement_type` like Damien said, if you look at the defect dojo api wrapper you see it is using `engagement_type="CI/CD"`

**Comment by valentijnscholten** [comment]
Started a poll to get feedback on what direction to take: https://owasp.slack.com/archives/C014H3ZV9U6/p1619887528113700

**Comment by mtesauro** [comment]
Restating my opinion from Slack here FWIW:
I'd keep the two hard coded options and remove the engagement_type model.  I think with the recent tagging improvements, you can get 90% of what having that separate model provides with far lower complexity.

I really like the idea of tags being something for 'location specific' customizations (e.g. what happens at **your** company) - it provides great flexibility without making the overall DefectDojo app keep to the general use case.

**Comment by damiencarol** [comment]
I checked the API v2 but  `engagement_type` is not in the filters for the `GET`. Oddly you can set it for the `POST` :D 
this is something we could fix in this release (`2.0.0`) I think.

**Comment by valentijnscholten** [comment]
fixed in dev/2.0.0 after 200 PRs sorry about that
